front/lib/swr/data_source_view_documents.ts

```diff
@@ -88,118 +50,94 @@ export function useDataSourceViewDocument({
   };
 }
 
-async function sendPatchRequest(
-  url: string,
-  {
-    arg,
-  }: {
-    arg: {
-      documentBody: PatchDataSourceWithNameDocumentRequestBody;
-    };
-  }
-) {
-  const res = await fetcherWithBody([url, arg.documentBody, "PATCH"]);
-  return res;
-}
-
-function decorateWithInvalidation<T>(
-  options: SWRMutationConfiguration<T, Error, string> | undefined,
-  invalidateCacheEntries: () => Promise<void>
-): SWRMutationConfiguration<T, Error, string> {
-  return options
-    ? {
-        ...options,
-        onSuccess: async (data, key, config) => {
-          await options.onSuccess?.(data, key, config);
-          await invalidateCacheEntries();
-        },
-      }
-    : {
-        onSuccess: invalidateCacheEntries,
-      };
-}
-
 export function useUpdateDataSourceViewDocument(
   owner: LightWorkspaceType,
   dataSourceView: DataSourceViewType,
-  documentName: string,
-  options?: SWRMutationConfiguration<PatchDocumentResponseBody, Error, string>
+  documentName: string
 ) {
-  // Used only for cache invalidation
-  const { mutate: mutateContentNodes } = useDataSourceViewContentNodes({
-    owner,
-    dataSourceView,
-    disabled: true,
-  });
+  const { mutateRegardlessOfQueryParams: mutateContentNodes } =
+    useDataSourceViewContentNodes({
+      owner,
+      dataSourceView,
+      disabled: true, // Needed just to create
+    });
 
   // Used only for cache invalidation
   const { mutateDocument } = useDataSourceViewDocument({
     owner,
     dataSourceView,
     documentId: documentName,
-    disabled: true,
+    disabled: true, // Needed just to create
   });
 
-  // Decorate options's onSuccess with cache invalidation
-  const invalidateCacheEntries = async () => {
-    await Promise.all([mutateContentNodes, mutateDocument]);
+  const sendNotification = useSendNotification();
+
+  const doUpdate = async (body: PatchDataSourceWithNameDocumentRequestBody) => {
+    const patchUrl = `/api/w/${owner.sId}/spaces/${dataSourceView.spaceId}/data_sources/${dataSourceView.dataSource.sId}/documents/${encodeURIComponent(documentName)}`;
+    const res = await fetch(patchUrl, {
```

Comment:
I think you created a better fetchWithBody which has the benefit to append the commit-hash, feel free to use (and update other places).